diff --git a/build/with-xcode-changes.js b/build/with-xcode-changes.js
index 343bcb9..ef65b0d 100644
--- a/build/with-xcode-changes.js
+++ b/build/with-xcode-changes.js
@@ -207,20 +207,15 @@ async function applyXcodeChanges(config, project, props) {
         }
     }
     if (targetToUpdate) {
+        const existingConfigurationList = targetToUpdate.props.buildConfigurationList;
         // Remove existing build phases
-        targetToUpdate.props.buildConfigurationList.props.buildConfigurations.forEach((config) => {
-            config.getReferrers().forEach((ref) => {
-                ref.removeReference(config.uuid);
+        if (existingConfigurationList) {
+            [...existingConfigurationList.props.buildConfigurations].forEach((config) => {
+                config.removeFromProject();
             });
-            config.removeFromProject();
-        });
-        // Remove existing build configuration list
-        targetToUpdate.props.buildConfigurationList
-            .getReferrers()
-            .forEach((ref) => {
-            ref.removeReference(targetToUpdate.props.buildConfigurationList.uuid);
-        });
-        targetToUpdate.props.buildConfigurationList.removeFromProject();
+            // Remove existing build configuration list
+            existingConfigurationList.removeFromProject();
+        }
         // Create new build phases
         targetToUpdate.props.buildConfigurationList =
             (0, configuration_list_1.createConfigurationListForType)(project, props);
